### PR TITLE
Thchang/ import and export shortcut in region list 1036

### LINKED ISSUE #1036: Import/export button in the region list widget
adding import and export shortcut buttons to the region list widget

### PR DESCRIPTION
**Pull request by TienHao**
closes #1036.

Added region import and export buttons in the region list widget.

### CODE REVIEW

**Comment by YuHsuan-Hwang** [comment]
The feature works nice
About the icons: in the file browser dialog and the z profile fitting widget, we have buttons for "exporting" files:
image
image
Shall we adjust these two buttons (or exchange the icons in the region list) for consistency?

**Comment by TienHao** [comment]
I think the "export" in file browser dialog and z profile fitting is to download files to user's computer, but the "export regions" is to store regions into server. Maybe we don't need to make these three export icons the same, but to change import(cloud: download) and export(cloud: upload) region button into these two icons. 
截圖 2021-08-13 下午2 22 45

**Comment by YuHsuan-Hwang** [comment]
I see the difference. Maybe we can change the locally download icons to those in line plots
image

**Comment by kswang1029** [comment]
@TienHao feature-wise this works well. Based on the current design, I would suggest we also add an export (all) region icon to the right of the import icon at the column head. If there is no region created, it is grey out. 

Just curious: have you considered adding just two buttons (import and export) to the bottom-right corner when mouse hovering over? That might be simpler. 🤔

**Comment by TienHao** [comment]
> @TienHao feature-wise this works well. Based on the current design, I would suggest we also add an export (all) region icon to the right of the import icon at the column head. If there is no region created, it is grey out.

nice suggestion. I have added.

> Just curious: have you considered adding just two buttons (import and export) to the bottom-right corner when mouse hovering over? That might be simpler.

Haven't though that before. The only advantage for the current design is to export inactive regions. I can change to hovered buttons. I think there might need 3 buttons(`import` , `export all`, `export active region`), or we just skip export all?

**Comment by kswang1029** [comment]
> > @TienHao feature-wise this works well. Based on the current design, I would suggest we also add an export (all) region icon to the right of the import icon at the column head. If there is no region created, it is grey out.
> 
> nice suggestion. I have added.
> 
> > Just curious: have you considered adding just two buttons (import and export) to the bottom-right corner when mouse hovering over? That might be simpler.
> 
> Haven't though that before. The only advantage for the current design is to export inactive regions. I can change to hovered buttons. I think there might need 3 buttons(`import` , `export all`, `export active region`), or we just skip export all?

The current design is fine. 👍 I was just thinking that since in the file browser we can select a subset of regions for export so we don’t need to show the export button for each region. But I can see the advantages of the current design so it is okay.

**Review by kswang1029 — APPROVED** [review]

**Review by YuHsuan-Hwang — APPROVED** [review]

File: src/components/RegionList/RegionListComponent.tsx [hunk 1]
```diff
@@ -89,7 +101,7 @@ export class RegionListComponent extends React.Component<WidgetProps> {
         let nameWidth = RegionListComponent.NAME_COLUMN_DEFAULT_WIDTH;
         const availableWidth = this.width - 2 * padding;
         let fixedWidth =
-            RegionListComponent.ACTION_COLUMN_DEFAULT_WIDTH * 2 +
+            RegionListComponent.ACTION_COLUMN_DEFAULT_WIDTH * 3 +
```
Comment (duidae) [inline src/components/RegionList/RegionListComponent.tsx]: pleasee help give constants meaningful definition like `OOO_WIDTH`.

File: src/components/RegionList/RegionListComponent.tsx [hunk 2]
```diff
@@ -192,7 +204,8 @@ export class RegionListComponent extends React.Component<WidgetProps> {
                 );
             } else {
                 lockEntry = (
-                    <td colSpan={2} style={{width: RegionListComponent.ACTION_COLUMN_DEFAULT_WIDTH * 2}}>
+                    <td colSpan={3} style={{width: RegionListComponent.ACTION_COLUMN_DEFAULT_WIDTH * 3}}>
```
Comment (duidae) [inline src/components/RegionList/RegionListComponent.tsx]: Same as above comment.

**Review by duidae — CHANGES_REQUESTED** [review]

File: src/components/RegionList/RegionListComponent.tsx (hunk 1 above)
Comment (TienHao) [reply src/components/RegionList/RegionListComponent.tsx]: has changed to `ACTIONS_COLUMN_DEFAULT_WIDTH`.

File: src/components/RegionList/RegionListComponent.tsx (hunk 2 above)
Comment (TienHao) [reply src/components/RegionList/RegionListComponent.tsx]: fixed.

**Review by duidae — APPROVED** [review]

**Pull request merged**